### PR TITLE
Hide/Show all toggle on MostFrequent components

### LINKED ISSUE #75: create action for HideAll button on MostFrequentComponent

### LINKED ISSUE #76: Be able to hide/show all conflated word's variants in MostUsedWords
ie most used word Lemon = [LeMON, lemon, LeMoN] 

currently can only hide/show the variants individual which means 3 clicks for user and 3 different actions

should be able to hide/show all variants with 1 click and 1 action since only advanced users are likely to care about the variants of the mostUsedWords

### PR DESCRIPTION
**Pull request by VasilyShelkov**
fixes #75 
fixed #76 

@Winwardo ready for comments + merge

### CODE REVIEW

File: src/client/results/Feed.js [hunk 1]
```diff
@@ -29,32 +36,78 @@ class Feed extends Component {
...
+  feed, toggledWords, wordsToggledAction, toggledUsers, usersToggledAction
+) => (
+  feed.filter((feedItem) => {
+    const content = feedItem.data.content;
+    const authorId = feedItem.author.id;
+
+    // If we can find the chosen hidden word in this tweet, block the post
+    for (const toggledWord of toggledWords) {
+      if (content.indexOf(toggledWord) > -1) {
+        return !wordsToggledAction && usersToggledAction;
```
Comment (Winwardo) [inline src/client/results/Feed.js]: I don't like "usersToggledAction" as a name. What is being toggled? Maybe "usersInvertHideAction" or something? I don't know, there has to be a more descriptive name

File: src/client/results/Feed.js [hunk 2]
```diff
@@ -29,32 +36,78 @@ class Feed extends Component {
...
+      }
+    }
+
+    for (const toggledUser of toggledUsers) {
+      if (authorId === toggledUser) {
+        return !usersToggledAction && wordsToggledAction;
```
Comment (Winwardo) [inline src/client/results/Feed.js]: Same for wordsToggledAction

**Comment by VasilyShelkov** [comment]
ok @Winwardo now ready for further comments + merge

File: src/client/results/mostfrequent/words/WordItemsList.js [hunk 3]
```diff
@@ -1,40 +1,90 @@
...
+              </div>
+            </div>
+            <div className="ten wide column">
+              {conflatedWordInfo.word} <i className="dropdown icon" />
+            </div>
           </div>
         </div>
-        <div className="fourteen wide column">
-          {conflatedWordInfo.word} <i className="dropdown icon" />
+        <div className="four wide column">
```
Comment (VasilyShelkov) [inline src/client/results/mostfrequent/words/WordItemsList.js:63]: Please @Winwardo suggest a cleaner way of doing the below if you think necessary, although this does work :P

File: src/client/results/Feed.js [hunk 4]
```diff
@@ -29,32 +36,78 @@ class Feed extends Component {
...
+const filterPostsForFeed = (
+  feed, toggledWords, isWordsToggledActionHide, toggledUsers, isUsersToggledActionHide
+) => (
+  feed.filter((feedItem) => {
+    const content = feedItem.data.content;
+    const authorId = feedItem.author.id;
+
+    // If we can find the chosen hidden word in this tweet, block the post
+    for (const toggledWord of toggledWords) {
+      if (content.indexOf(`${toggledWord}`) > -1) {
```
Comment (Winwardo) [inline src/client/results/Feed.js]: Any reason this isn't just `toggledWord` directly rather than wrapping in backticks?

File: src/client/results/mostfrequent/mostFrequentActions.spec.js [hunk 5]
```diff
@@ -34,4 +36,16 @@ describe('#MostFrequentActions', () => {
       userId: '12345',
     });
   });
+
+  it('should create an action for toggling all most used words', () => {
+    toggleAllMostUsedWordsSearch('12345').should.deep.equal({
```
Comment (Winwardo) [inline src/client/results/mostfrequent/mostFrequentActions.spec.js]: I thought you meant to remove the 12345 parameter?

**Comment by Winwardo** [comment]
I understand it's hard to break into useful functions, so could you comments MostUsedWords and WordItemsList? It's very hard to follow what's going on, some clarification would really go a long way

File: src/client/results/Feed.js (hunk 4 above)
Comment (VasilyShelkov) [inline src/client/results/Feed.js]: It was because I tried it with having spaces either side :P will change

File: src/client/results/mostfrequent/mostFrequentActions.spec.js (hunk 5 above)
Comment (VasilyShelkov) [inline src/client/results/mostfrequent/mostFrequentActions.spec.js]: lol missed it last night because you didn't leave a comment X)

File: src/client/results/mostfrequent/words/WordItemsList.js [hunk 6]
```diff
@@ -62,7 +62,12 @@ const ConflatedWordItem = ({ accordianIndex, toggleMostUsedWords, conflatedWordI
         </div>
         <div className="four wide column">
           <div data-id={`${conflatedWordInfo.word}master`} className="ui checkbox words" onClick={() => {
+            // makes sure the accordian is open
```
Comment (Winwardo) [inline src/client/results/mostfrequent/words/WordItemsList.js]: *accordion
(there are a few instances of this)

File: src/client/results/mostfrequent/words/WordItemsList.js [hunk 7]
```diff
@@ -62,7 +62,12 @@ const ConflatedWordItem = ({ accordianIndex, toggleMostUsedWords, conflatedWordI
...
+            // so that the user can see which words are being toggled
             $('.ui.accordion.words').accordion('open', accordianIndex);
+
+            // sets the semantic-ui checkbox action method dependant on whether
```
Comment (Winwardo) [inline src/client/results/mostfrequent/words/WordItemsList.js]: *dependent

File: src/client/results/mostfrequent/words/WordItemsList.js [hunk 8]
```diff
@@ -118,6 +130,8 @@ const WordItem = ({ toggleMostUsedWords, makeupInfo, conflatedWord, checkboxId,
       <div data-id={`${conflatedWord}child${checkboxId}`} className="ui checkbox words" onClick={() => {
         toggleMostUsedWords([makeupInfo.word]);
         const $thisCheckbox = $(`.ui.checkbox.words[data-id^="${conflatedWord}child${checkboxId}"]`).checkbox('is checked');
+        // sets the semantic-ui checkbox action method dependant on whether
```
Comment (Winwardo) [inline src/client/results/mostfrequent/words/WordItemsList.js]: dependent
